ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py

```diff
@@ -1268,3 +1268,106 @@ def test_rfftn(
         axes=axes,
         norm=norm,
     )
+
+
+# stft
+@st.composite
+def stft_arguments(draw):
+    dtype = draw(helpers.get_dtypes("float"))
+    n_fft_type = draw(st.sampled_from(["int", "tuple"]))
+
+    if n_fft_type == "int":
+        n_fft = draw(st.integers(min_value=2, max_value=256))
+    else:
+        n_fft = tuple(draw(st.integers(min_value=2, max_value=256)) for _ in range(2))
+
+    hop_length = draw(st.integers(min_value=1, max_value=256))
+    axis = draw(st.integers(min_value=0))
+    onesided = draw(st.booleans())
+    fs = draw(st.floats(min_value=0.001, max_value=1.0))
+    window = "hann"
+    win_length = draw(st.integers(min_value=1, max_value=n_fft))
+    noverlap = draw(st.integers(min_value=0, max_value=win_length - 1))
+    center = draw(st.booleans())
+    pad_mode = draw(st.sampled_from(["reflect", "constant"]))
+    normalized = draw(st.booleans())
+    detrend = draw(st.one_of(st.booleans(), st.sampled_from(["linear", "constant"])))
+    return_complex = draw(st.booleans())
+    boundary = draw(st.sampled_from(["reflect", "zeros"]))
+    return (
+        dtype,
+        n_fft,
+        hop_length,
+        axis,
+        onesided,
+        fs,
+        window,
+        win_length,
+        noverlap,
+        center,
+        pad_mode,
+        normalized,
+        detrend,
+        return_complex,
+        boundary,
+    )
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.stft",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+    ),
+    ground_truth_backend="numpy",
+    test_gradients=st.just(False),
+    stft_args=stft_arguments(),
+)
+def test_stft(
```

Comment:
Hey, could you please fix tests: `hypothesis.errors.InvalidArgument: max_value=(2, 2) must be a real number.`